gtk/Session.cc

```diff
@@ -308,13 +308,30 @@ void Session::Impl::dec_busy()
 namespace
 {
 
-bool is_valid_eta(int t)
+template<typename T>
+// NOLINTNEXTLINE(bugprone-easily-swappable-parameters)
+constexpr int compare_generic(T const& a, T const& b)
```

Comment:
I'm not convinced we need `constexpr` here and below. Sure, these functions can be used at compile time, but they never will be since they're only used for torrents sorting that always happens at runtime. Is there a difference in code generation that I'm not seeing?

Reply:
I've been adding `constexpr` to functions where possible to tighten the function's contract, i.e. for benefits in the compiler checks rather than the compiler output.

I haven't compared the outputs in Compiler Explorer to see if the generated code is any different.

Reply:
Any publications you could share on how this helps catch issues (or in any other way)?

Reply:
Some of the benefits are intrinsic to the language definition of `constexpr`: constexpr functions cant' have try-catch or asm or goto, that there are no uninitialized variables, no thread-storage or static variable declarations, and that in C++17 there's no memory allocation and in C++20 it's bounded by transient allocation.

But more of it is also from knowing what `constexpr` can't call -- I know it's not going to be calling system functions.

TBF, we're also using clang-tidy to ensure there are no gotos or assembly or uninitialized variables.

Reply:
Yeah, I was more interested in research of cases like "this bug wouldn't have been there if function was marked `constexpr`"...